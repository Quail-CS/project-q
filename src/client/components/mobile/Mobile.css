.mobile-takeover {
  height: 100vh;
  width: 100vw;

  display: flex;
  flex-direction: column
}

.mobile-takeover-image {
  padding: 10vh 10vw 5vh 10vw;
  height: auto;
  max-height: 20vh;
  max-width: 80vw;
  width: auto;
}

@media only screen and (orientation: landscape) {
  .mobile-takeover-image {
    padding: 10vh 30vw 5vh 30vw;
    height: auto;
    max-height: 20vh;
    max-width: 40vw;
    width: auto;
  }
}

.mobile-takeover-text {
  color: #FFFFFF;
  font-size: 5rem;
  padding: 5vh 10vw;
  text-align: center;
}