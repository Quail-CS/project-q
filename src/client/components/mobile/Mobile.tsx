import * as React from 'react';

const brandingLogo = require('../../assets/brandingLogo.png')
import './Mobile.css';

const Mobile = () => {
  return (
    <div className="mobile-takeover">
      <img alt="Welcome to eTrade powered by Ethiq" className="mobile-takeover-image" src={brandingLogo} />
      <div className="mobile-takeover-text">Ethiq Beta Is Currently Optimized For Desktop</div>
    </div>
  )
};

export default Mobile;
