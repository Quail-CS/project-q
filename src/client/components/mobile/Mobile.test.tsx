import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import Mobile from './Mobile';

Enzyme.configure({ adapter: new Adapter() });

describe('Mobile Component', () => {
  it('matches the snapshot', () => {
    const tree: Renderer.ReactTestRendererJSON = Renderer.create(<Mobile />).toJSON() as Renderer.ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  });
  it('renders a <div> with class mobile-takeover', () => {
    const mobileWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Mobile />);
    expect(mobileWrapper.type()).toEqual('div');
    expect(mobileWrapper.hasClass('mobile-takeover')).toEqual(true);
  });
  it('contains 1 <img> with class mobile-takeover-image', () => {
    const mobileWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Mobile />);
    const mobileImg = mobileWrapper.find('.mobile-takeover-image');
    expect(mobileImg).toHaveLength(1);
    expect(mobileImg.type()).toEqual('img');
  });
  it('contains 1 <div> with class mobile-takeover-text with text "Ethiq Beta Is Currently Optimized For Desktop', () => {
    const mobileWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Mobile />);
    const mobileText = mobileWrapper.find('.mobile-takeover-text');
    expect(mobileText).toHaveLength(1);
    expect(mobileText.type()).toEqual('div');
    expect(mobileText.text()).toEqual('Ethiq Beta Is Currently Optimized For Desktop');
  });
});
