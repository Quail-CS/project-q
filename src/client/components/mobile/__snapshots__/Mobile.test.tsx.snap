// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Component matches the snapshot 1`] = `
<div
  className="mobile-takeover"
>
  <img
    alt="Welcome to eTrade powered by Ethiq"
    className="mobile-takeover-image"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="mobile-takeover-text"
  >
    Ethiq Beta Is Currently Optimized For Desktop
  </div>
</div>
`;
